src/main/java/teammates/common/util/FieldValidator.java

```diff
@@ -417,12 +418,11 @@ public String getInvalidityInfoForStudentRoleComments(String studentRoleComments
      * @return An explanation of why the {@code gender} is not acceptable.
      *         Returns an empty string if the {@code gender} is acceptable.
      */
-    public String getInvalidityInfoForGender(String gender) {
+    public String getInvalidityInfoForGender(GenderType gender) {
```

Comment:
Since GenderType always store valid genderType: male, female or other - you can remove this method and related variables like GENDER_ERROR_MESSAGE and GENDER_ACCEPTED_VALUES